authorization/role/repository/default_role_mapping.go

```diff
@@ -0,0 +1,202 @@
+	ToRoleID uuid.UUID
+}
+
+func (m DefaultRoleMapping) TableName() string {
+	return "default_role_mapping"
+}
+
+// GetLastModified returns the last modification time
+func (m DefaultRoleMapping) GetLastModified() time.Time {
+	return m.UpdatedAt
+}
+
+// GormDefaultRoleRepository is the implementation of the storage interface for Role.
+type GormDefaultRoleMappingRepository struct {
+	db *gorm.DB
+}
+
+// NewDefaultRoleMappingRepository creates a new storage type.
+func NewDefaultRoleMappingRepository(db *gorm.DB) DefaultRoleMappingRepository {
+	return &GormDefaultRoleMappingRepository{db: db}
+}
+
+// DefaultRoleMappingRepository represents the storage interface.
+type DefaultRoleMappingRepository interface {
+	CheckExists(ctx context.Context, ID uuid.UUID) (bool, error)
+	Load(ctx context.Context, ID uuid.UUID) (*DefaultRoleMapping, error)
+	Create(ctx context.Context, u *DefaultRoleMapping) error
+	Save(ctx context.Context, u *DefaultRoleMapping) error
+	List(ctx context.Context) ([]DefaultRoleMapping, error)
+	Delete(ctx context.Context, ID uuid.UUID) error
+	FindForResourceType(ctx context.Context, resourceTypeID uuid.UUID) ([]DefaultRoleMapping, error)
+}
+
+// TableName overrides the table name settings in Gorm to force a specific table name
+// in the database.
+func (m *GormDefaultRoleMappingRepository) TableName() string {
+	return "default_role_mapping"
+}
+
+// CheckExists returns nil if the given ID exists otherwise returns an error
+func (m *GormDefaultRoleMappingRepository) CheckExists(ctx context.Context, ID uuid.UUID) (bool, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "default_role_mapping", "exists"}, time.Now())
+
+	var exists bool
+	query := fmt.Sprintf(`
+		SELECT EXISTS (
+			SELECT 1 FROM %[1]s
+			WHERE
+				default_role_mapping_id=$1
+				AND deleted_at IS NULL
+		)`, m.TableName())
+
+	err := m.db.CommonDB().QueryRow(query, ID.String()).Scan(&exists)
+	if err == nil && !exists {
+		return exists, errors.NewNotFoundError(m.TableName(), ID.String())
+	}
+	if err != nil {
+		return false, errors.NewInternalError(ctx, errs.Wrapf(err, "unable to verify if %s exists", m.TableName()))
+	}
+	return exists, nil
+}
+
+// CRUD Functions
+
+// Load returns a single RoleMapping as a Database Model
+func (m *GormDefaultRoleMappingRepository) Load(ctx context.Context, id uuid.UUID) (*DefaultRoleMapping, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "default_role_mapping", "load"}, time.Now())
+	var native DefaultRoleMapping
+	err := m.db.Table(m.TableName()).
+		Preload("ResourceType").
+		Preload("FromRole").
+		Preload("ToRole").
+		Where("default_role_mapping_id = ?", id).Find(&native).Error
+	if err == gorm.ErrRecordNotFound {
+		return nil, errors.NewNotFoundError("default_role_mapping", id.String())
+	}
+	return &native, errs.WithStack(err)
+}
+
+// Create creates a new record.
+func (m *GormDefaultRoleMappingRepository) Create(ctx context.Context, u *DefaultRoleMapping) error {
```

Comment:
Please add more details to the method comment about the default role mapping object fields. So, it's clear how to create a default role mapping. Some example would be great.
Maybe we should add this description to the Default Role Mapping object itself.

Reply:
I've added quite a large comment to the DefaultRoleMapping struct